queries.sql

```diff
@@ -113,3 +113,72 @@ JOIN animals a ON o.id = a.owner_id
 GROUP BY o.full_name
 ORDER BY animal_count DESC
 LIMIT 1;
+
+--Vet clinic database: add "join table" for visits exercise:
+SELECT a.name AS animal_name
+FROM animals a
+INNER JOIN visits v ON a.id = v.animal_id
+INNER JOIN vets ve ON ve.id = v.vet_id
+WHERE ve.name = 'William Tatcher'
+ORDER BY v.visit_date DESC
+LIMIT 1;
+
+SELECT COUNT(DISTINCT v.animal_id) AS num_animals
+FROM visits v
+INNER JOIN vets ve ON ve.id = v.vet_id
+WHERE ve.name = 'Stephanie Mendez';
+
+SELECT v.name AS vet_name, s.name AS specialty_name
+FROM vets v
+LEFT JOIN specializations sp ON v.id = sp.vet_id
+LEFT JOIN species s ON sp.species_id = s.id;
+
+SELECT a.name AS animal_name
+FROM animals a
+INNER JOIN visits v ON a.id = v.animal_id
+INNER JOIN vets ve ON ve.id = v.vet_id
+WHERE ve.name = 'Stephanie Mendez'
+  AND v.visit_date BETWEEN '2020-04-01' AND '2020-08-30';
+
+SELECT a.name AS animal_name, COUNT(v.animal_id) AS visit_count
+FROM animals a
+INNER JOIN visits v ON a.id = v.animal_id
+GROUP BY a.name
+ORDER BY visit_count DESC
+LIMIT 1;
+
+SELECT a.name AS animal_name, v.visit_date
+FROM animals a
+INNER JOIN visits v ON a.id = v.animal_id
+INNER JOIN vets ve ON ve.id = v.vet_id
+WHERE ve.name = 'Maisy Smith'
+ORDER BY v.visit_date ASC
+LIMIT 1;
+
+SELECT a.name AS animal_name, v.visit_date, ve.name AS vet_name
+FROM animals a
+INNER JOIN visits v ON a.id = v.animal_id
+INNER JOIN vets ve ON ve.id = v.vet_id
+ORDER BY v.visit_date DESC
+LIMIT 1;
+
+SELECT COUNT(*) AS num_visits
+FROM visits v
+INNER JOIN vets ve ON ve.id = v.vet_id
+INNER JOIN specializations s ON ve.id = s.vet_id
+INNER JOIN species sp ON sp.id = s.species_id
+WHERE sp.id != v.animal_id;
+
+
```

Comment:
![image](https://github.com/fpsapc/vetClinic/assets/93039359/5436ea7a-decb-41b1-ba04-4abe38df4d3c)

- [ ] It would be nice if after updating your database, you make changes to this query as well because this query should return 12 records but now it is returning only 9 🤔 Kindly make sure if the data is the only problem for the wrong result of this query or the query itself is giving issues as well 👍

Reply:
Hello, @HAMMAS-SALEEM: Thanks for your kind review, I just make the required changes. Here is the screenshot of updated database and required query.
![CORRECTIONS](https://github.com/fpsapc/vetClinic/assets/112868013/380fb950-ddf7-4a63-b1ce-eedc43934f98)